Terminal.Gui/Drawing/SixelEncoder.cs

```diff
@@ -0,0 +1,224 @@
+﻿// This code is based on existing implementations of sixel algorithm in MIT licensed open source libraries
+// node-sixel (Typescript) - https://github.com/jerch/node-sixel/tree/master/src
+// Copyright (c) 2019, Joerg Breitbart @license MIT
+// libsixel (C/C++) - https://github.com/saitoha/libsixel
+// Copyright (c) 2014-2016 Hayaki Saito @license MIT
+
+using Terminal.Gui;
+
+namespace Terminal.Gui;
+
+/// <summary>
+/// Encodes a images into the sixel console image output format.
+/// </summary>
+public class SixelEncoder
+{
+    /*
+
+    A sixel is a column of 6 pixels - with a width of 1 pixel
+
+    Column controlled by one sixel character:
+      [ ]  - Bit 0 (top-most pixel)
+      [ ]  - Bit 1
+      [ ]  - Bit 2
+      [ ]  - Bit 3
+      [ ]  - Bit 4
+      [ ]  - Bit 5 (bottom-most pixel)
+
+   Special Characters
+       The '-' acts like '\n'. It moves the drawing cursor
+       to beginning of next line
+
+       The '$' acts like the <Home> key.  It moves drawing
+       cursor back to beginning of the current line
+       e.g. to draw more color layers.
+
+   */
+
+
+    /// <summary>
+    /// Gets or sets the quantizer responsible for building a representative
+    /// limited color palette for images and for mapping novel colors in
+    /// images to their closest palette color
+    /// </summary>
+    public ColorQuantizer Quantizer { get; set; } = new ();
+
+    /// <summary>
+    /// Encode the given bitmap into sixel encoding
+    /// </summary>
+    /// <param name="pixels"></param>
+    /// <returns></returns>
+    public string EncodeSixel (Color [,] pixels)
+    {
+
+        const string start = "\u001bP"; // Start sixel sequence
+
+        const string defaultRatios = "0;0;0"; // Defaults for aspect ratio and grid size
```

Comment:
@tznind If you're trying to get transparent images to work, you need to set the second parameter here to 1.

Reply:
Wow thank you so much, I Would never have realised that.  As soon as I changed to 1 it started working.

Do you know how to get the pixels to runes ratio from the terminal? is there an ASCII terminal sequence I can send to get that info (When sixel supported)?

For Windows Terminal preview it seems to be 10 horizontal and 20 vertical.  This is needed to constrain the sixel rendered to match up with the View it is presented in.

Reply:
@tznind You can query the size in pixels of the character cell (which I assume is what you're referring to as the "runes") with `CSI 16 t`. For terminals that don't support that, you can query the window size in pixels with `CSI 14 t`, and the window size in characters with `CSI 18 t`, and divide the one by the other to obtain the size of the character cell.

But be aware that the latter calculation won't always be exact (some terminals will include things like the border and title as part of the window size), so you may need to round the result. Also note that real Sixel terminals won't support those queries, so 10x20 is a reasonable fallback if you don't get any query responses.